rpcd/playbooks/roles/rpc_post_upgrade/tasks/post-upgrade-utility.yml

```diff
@@ -70,3 +70,21 @@
     msg: "One or more of the cinder services are in the down state. Check the output above for more detail."
   when: "'down' in '{{ item.split(' ')[-1] }}'"
   with_items: "{{ cinder_output.stdout_lines | default([]) }}"
+
+# Adds orchestration service checks per https://rpc-openstack.atlassian.net/browse/UG-543
+- name: Grab relevant output of orchestration service list
+  shell: |
+    . ~/openrc
+    openstack orchestration service list --insecure
+  register: orchestration_output
+  failed_when: "orchestration_output.rc != 0"
+
+- name: Display output of orchestration service list
+  debug:
+    var: orchestration_output
+
+- name: Fail if any of the orchestration services are down
+  fail:
+    msg: "One or more of the orchestration services are down. Check the output above for more detail."
+  when: "'down' in '{{ item.split(' ')[-1] }}'"
```

Comment:
When testing for errors I think it's better to make the validation a little more exacting, e.g.

```yaml
- name: Grab relevant output of orchestration service list
  shell: |
    . ~/openrc
    openstack orchestration service list --insecure  -f json
  register: orchestration_output
  failed_when: "orchestration_output.rc != 0"

- name: Fail if any of the orchestration services are down
  fail:
    msg: "One or more of the orchestration services are down. Check the output above for more detail."
  when: item["status"] != "up"
  with_items: "{{ orchestration_output | from_json }}"
```

By using JSON the specific piece of data, `status`, can be referenced instead of having to rely on ordering or successful text parsing. By testing for inequality with `"up"` it protects against changes to status names as well as the unexpected.